docs/doxygen/collectives.md

```diff
@@ -1,20 +1,150 @@
 ## Starting and Completing the Collectives
 
-A UCC collective operation is a group communication operation among the participants of the team. All participants of the team are required to call the collective operation. Each participant is represented by the endpoint that is unique to the team used for the collective operation. This section provides a set of routines for launching, progressing, and completing the collective operations. 
+A UCC collective operation is a group communication operation among the
+participants of the team. All participants of the team are required to call the
+collective operation. Each participant is represented by the endpoint that is
+unique to the team used for the collective operation. This section provides a
+set of routines for launching, progressing, and completing the collective
+operations.
 
-\b Invocation semantics: The ucc\_collective\_init routine is a non-blocking collective operation to initialize the buffers, operation type, reduction type, and other information required for the collective operation. All participants of the team should call the initialize operation. The routine returns once the participants enter the collective initialize operation. The collective operation is invoked using a ucc\_collective\_post operation. ucc\_collective\_init\_and\_post operation initializes as well as post the collective operation.
+**Invocation semantics**: The ucc\_collective\_init routine is a non-blocking
+collective operation to initialize the buffers, operation type, reduction type,
+and other information required for the collective operation. All participants of
```

Comment:
_The routine returns once the participants enter the collective initialize operation._

This is a strong statement - i never realized we have such a constraint. Does this mean that ucc_collective_init is potentially blocking, ie it says that the call "may not return" until everybody enter collective. I think this is incorrect.